webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/GroupService.scala

```diff
@@ -5,12 +5,62 @@
 
 package org.knora.webapi.slice.admin.domain.service
 
-import zio.ZLayer
+import zio._
+
+import dsp.errors.InconsistentRepositoryDataException
+import dsp.errors.NotFoundException
+import org.knora.webapi.slice.admin.domain.model.Group
+import org.knora.webapi.slice.admin.domain.model.GroupDescriptions
+import org.knora.webapi.slice.admin.domain.model.GroupIri
+import org.knora.webapi.slice.admin.domain.model.GroupName
+import org.knora.webapi.slice.admin.domain.model.GroupSelfJoin
+import org.knora.webapi.slice.admin.domain.model.GroupStatus
+import org.knora.webapi.slice.admin.domain.model.KnoraGroup
+import org.knora.webapi.slice.admin.domain.model.KnoraProject.ProjectIri
 
 final case class GroupService(
   private val knoraGroupService: KnoraGroupService,
-) {}
+  private val projectService: ProjectService,
+) {
+  def findAll: Task[List[Group]] = knoraGroupService.findAll.flatMap(ZIO.foreachPar(_)(toGroup))
+
+  def findById(id: GroupIri): Task[Option[Group]] = knoraGroupService.findById(id).flatMap(ZIO.foreach(_)(toGroup))
+
+  private def toGroup(knoraGroup: KnoraGroup): Task[Group] =
+    for {
+      projectIri <- ZIO.fromOption(knoraGroup.belongsToProject).orElseFail(NotFoundException("Project IRI not found."))
```

Comment:
**issue**: `Group` and `KnoraGroup` should be aligned with the matching properties having the exaclty same cardinality

This seems like a weird mismatch between `Group` and `KnoraGroup`. IMHO, both should either always require a project or both should make the project optional.

Can you please look into why not both require a project they belong to? This has to be fixed.

Reply:
`Group` formerly know as `GroupADM` has it as implemented like that since forever. The `KnoraGroup` was introduced recently and it has the optional project, I think due to in-built groups which are not tided with any projects by default. The same situation is with `User` and `KnoraUser` and their password, see [here](https://github.com/dasch-swiss/dsp-api/blob/9445614b984938a919c4ba986194c45bc906537e/integration/src/test/scala/org/knora/webapi/e2e/admin/GroupsADME2ESpec.scala). I'm confused now how this should look like.

Reply:
The admin ontology allows for a group not belonging to a project. 

I would suggest to die here and log an error.